test/hyperbahn-client/hostports.js

```diff
@@ -268,18 +269,8 @@ function runTests(HyperbahnCluster) {
             onResponse
         );
         function onResponse(err, res) {
-            assert.ok(err, 'should be error');
-            assert.equals(err.type, 'tchannel-thrift-handler.parse-error.body-failed',
-                'error should be parser error');
-            assert.ok(err.message.indexOf(
-                'Could not parse body (arg3) argument.\n' +
-                'Expected Thrift encoded arg3 for endpoint Hyperbahn::discover.') === 0,
-                'error message should be a parsing failure');
-
-            var items = cluster.logger.items();
-            assert.ok(items.length > 0 && items[0].msg === 'Got unexpected invalid thrift for arg3',
-                'Do not miss the error log');
-
+            // TODO expect UnrecognizedException
+            assert.ok(err || !res.ok, 'should be error (possibly unrecognized application error)');
```

Comment:
we will have to revisit when unrecognized exceptions are implemented. atm, you get a pretty bad middle ground.